GEOSldas_App/ldas_setup

```diff
@@ -152,32 +155,79 @@ class LDASsetup:
-           self.rqdExeInp[ 'MET_TAG' ] = self.adas_expid + '__bkg'
+           self.adas_expdir                           = os.path.dirname(self.exphome)
+           self.rqdExeInp['ADAS_EXPDIR']              = self.adas_expdir
+           self.adas_expid                            = os.path.basename(self.adas_expdir)
+           self.rqdExeInp['MET_TAG']                  = self.adas_expid + '__bkg'
            if self.ladas_coupling == 1 :
-                 self.rqdExeInp[ 'EXP_ID' ] = self.adas_expid + '_LDAS'
-           if self.ladas_coupling == 2 :
-                 self.rqdExeInp[ 'EXP_ID' ] = self.adas_expid + '_LDAS4ens'
-                 self.rqdExeInp[ 'MET_PATH' ] = self.adas_expdir +'/atmens/mem'
-
-        self.first_ens_id = int(self.rqdExeInp.get('FIRST_ENS_ID',0))
+               # ldas coupled with determistic component of adas
+               self.rqdExeInp['EXP_ID']               = self.adas_expid  + '_LDAS'
+               self.rqdExeInp['MET_PATH']             = self.adas_expdir + '/recycle/holdpredout'
+               self.rqdExeInp['ENSEMBLE_FORCING']     = 'NO'
+               self.nens = 24
+           elif self.ladas_coupling == 2 :
+               # ldas coupled with ensemble component of adas
+               self.rqdExeInp['EXP_ID']               = self.adas_expid  + '_LDAS4ens'
+               self.rqdExeInp['MET_PATH']             = self.adas_expdir + '/atmens/mem'
+               self.rqdExeInp['ENSEMBLE_FORCING']     = 'YES' 
+               self.nens = 32
+           else :
+               exit("Error. Unknown value of ladas_coupling.\n")
+           self.rqdExeInp['NUM_LDAS_ENSEMBLE']        = self.nens
+           self.first_ens_id                          = 1
+           self.rqdExeInp['FIRST_ENS_ID']             = self.first_ens_id 
+           self.rqdExeInp['EXP_DOMAIN']               = 'CF'+ self.adasdomain +'x6C_GLOBAL'
+           self.rqdExeInp['BCS_RESOLUTION']           = 'CF'+ self.adasdomain +'x6C_CF'     + self.adasdomain +'x6C'
+           self.rqdExeInp['RESTART_DOMAIN']           = 'CF'+ self.adasdomain +'x6C_GLOBAL' 
+           self.rsdir                                 = os.path.dirname(self.rstdir)
+           self.rqdExeInp[ 'RESTART_PATH' ]           = self.rsdir 
+           self.rqdExeInp[ 'RESTART_ID'   ]           = os.path.basename(self.rstdir)
+           # the following are not in default rqdExeInp list 
+           self.met_hinterp                           = int(self.rqdExeInp.get('MET_HINTERP',0))
+           self.rqdExeInp['MET_HINTERP']              = 0
+           self.landassim_dt                          = str(self.rqdExeInp.get('LANDASSIM_DT', "10800"))
+           self.rqdExeInp['LANDASSIM_DT']             = "10800"
+           self.landassim_t0                          = str(self.rqdExeInp.get('LANDASSIM_T0', "013000"))
+           self.rqdExeInp['LANDASSIM_T0']             = "013000"
+           jobsgmtdef                                 = "00000000 060000"
```

Comment:
@saraqzhang , @weiyuan-jiang : Do we need `jobsgmtdef` and `self.jobsgmt` ? Can we just use
`self.rqdExeInp['JOB_SGMT'] = "00000000 060000"`

Reply:
edited